sdk/core/azure-core/src/main/java/com/azure/core/util/polling/PollerFlux.java

```diff
@@ -87,19 +82,103 @@ public PollerFlux(Duration defaultPollInterval,
                 "Negative or zero value for 'defaultPollInterval' is not allowed."));
         }
         this.defaultPollInterval = defaultPollInterval;
-        this.activationOperation = Objects.requireNonNull(activationOperation,
-            "'activationOperation' cannot be null.");
-        this.oneTimeActivationMono = oneTimeActivationMono(activationOperation);
+        Objects.requireNonNull(activationOperation, "'activationOperation' cannot be null.");
         this.pollOperation = Objects.requireNonNull(pollOperation, "'pollOperation' cannot be null.");
         this.cancelOperation = Objects.requireNonNull(cancelOperation, "'cancelOperation' cannot be null.");
         this.fetchResultOperation = Objects.requireNonNull(fetchResultOperation,
             "'fetchResultOperation' cannot be null.");
+        this.oneTimeActivationMono = new OneTimeActivation<>(this.rootContext,
+            activationOperation,
+            // mapper
+            activationResult -> new PollResponse<>(LongRunningOperationStatus.NOT_STARTED, activationResult)).getMono();
+        this.syncActivationOperation =
+            cxt -> new PollResponse<>(LongRunningOperationStatus.NOT_STARTED, activationOperation.apply(cxt).block());
+    }
+
+    /**
+     * Creates PollerFlux.
+     *
+     * PollerFlux obtained from this factory method uses an activationOperation which returns a Mono that
+     * emits {@link PollResponse}, which holds the result.
+     * The PollerFlux created from constructor uses an activationOperation which returns a Mono that directly
+     * emits result.
+     * Since the first variant of PollerFlux has access to the response, it can skip the polling loop if the
+     * response indicate that LRO is completed. Whereas the second PollerFlux variant calls pollFunction at
+     * least once.
+     *
+     * @param defaultPollInterval the default polling interval
+     * @param activationOperation the activation operation to be invoked at most once across all subscriptions,
+     *                            this parameter is required, if there is no specific activation work to be
+     *                            done then invocation should return Mono.empty(), this operation will be called
+     *                            with a new {@link PollingContext}.
```

Comment:
Don't worry about this kind of indentation - checkstyle should be warning against this too - please check if checkstyle passes.

Reply:
This is a very long, single sentence. Break it down into a few sentences.